libcontainer/process_linux.go

```diff
@@ -72,15 +72,15 @@ func (p *setnsProcess) start() (err error) {
 	p.childPipe.Close()
 	p.rootDir.Close()
 	if err != nil {
-		return newSystemErrorWithCause(err, "starting setns process")
+		return parentError(p.parentPipe, err, "starting setns process")
 	}
 	if p.bootstrapData != nil {
 		if _, err := io.Copy(p.parentPipe, p.bootstrapData); err != nil {
-			return newSystemErrorWithCause(err, "copying bootstrap data to pipe")
+			return parentError(p.parentPipe, err, "copying bootstrap data to pipe")
```

Comment:
For all of these, how can you be sure that there is a pending error on `p.parentPipe` or the process had died? Otherwise `.Decode` in `parseSync` will block, no?